### PR TITLE
Feature: Dropout Student Handling

### PR DESCRIPTION
**Pull request by Bahugunajii**
### ADR
https://www.notion.so/avantifellows/ADR-Dropout-Students-Handling-b31ca1a7e5574df996798e8a5a63de44

### Description
- Added `status` schema, migrations, controller, and view
- Updated `student` view
- Added `dropout` endpoint
- Added logic to remove all the `group-user` entries
- Added logic to create new ER for dropout students
- Added logic to update the `status` field in student to `dropout`

### Checklist
- [x] Self-testing

### CODE REVIEW

File: lib/dbservice/users/student.ex [hunk 1]
```diff
@@ -48,6 +48,7 @@ defmodule Dbservice.Users.Student do
     field(:grade_10_marksheet, :string)
     field(:photo, :string)
     field(:planned_competitive_exams, {:array, :integer})
+    field(:status, :string, default: "registered")
```
Comment (suryabulusu) [inline lib/dbservice/users/student.ex:51]: why not connecting with status table

File: lib/dbservice_web/controllers/student_controller.ex [hunk 2]
```diff
@@ -158,4 +163,72 @@ defmodule DbserviceWeb.StudentController do
...
+    # Check if the student's status is already 'dropout'
+    if student.status == "dropout" do
+      conn
+      |> put_status(:bad_request)
+      |> json(%{error: "Student is already marked as dropout"})
+    else
+      user_id = student.user_id
+      current_time = DateTime.utc_now()
+
+      # Fetch status and group details in a single query
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/student_controller.ex:179]: why set group type when we already know it is gonna be "status"

File: lib/dbservice_web/controllers/student_controller.ex [hunk 3]
```diff
@@ -158,4 +163,72 @@ defmodule DbserviceWeb.StudentController do
...
+          is_current: false,
+          end_date: current_time
+        })
+
+        # Create a new enrollment record with the fetched group_id
+        new_enrollment_attrs = %{
+          user_id: user_id,
+          is_current: true,
+          start_date: current_time,
+          group_id: group_id,
```
Comment (suryabulusu) [inline lib/dbservice_web/controllers/student_controller.ex:210]: what would this group_id refer to? where group_type is status?

File: lib/dbservice_web/swagger_schemas/status.ex [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        swagger_schema do
+          title("Status")
+          description("A status in the application")
+
+          properties do
+            title(:string, "Title of the status")
+          end
+
+          example(%{
+            title: "Title of an enrollment record"
```
Comment (suryabulusu) [inline lib/dbservice_web/swagger_schemas/status.ex:18]: title: registered

File: lib/dbservice/users/student.ex (hunk 1 above)
Comment (Bahugunajii) [reply lib/dbservice/users/student.ex:51]: Just for fast querying

File: lib/dbservice_web/controllers/student_controller.ex (hunk 2 above)
Comment (Bahugunajii) [reply lib/dbservice_web/controllers/student_controller.ex:179]: We are fetching the `group` here where `type` will is `status`. So basically not setting anything & just fetching. XD

File: lib/dbservice_web/controllers/student_controller.ex (hunk 3 above)
Comment (Bahugunajii) [reply lib/dbservice_web/controllers/student_controller.ex:210]: `group_id` is the ID (PK) of `group`. Each record have `type` and `child_id` in it. That `child_id` will refer PK of `status` table

File: lib/dbservice_web/swagger_schemas/status.ex (hunk 4 above)
Comment (Bahugunajii) [reply lib/dbservice_web/swagger_schemas/status.ex]: Updated!

**Review by suryabulusu — APPROVED** [review]
looks good

**Pull request merged**